#### Задача 1.

Напишите собственный сегментатор по предложениям. Подумайте о том, что вы хотите считать за границы предложения. 

In [ ]:
# your code here

#### Задача 2.

Напишите свой токенизатор. Это должна быть функция, которая принимает строку текста и возвращает список токенов. Считаем, что в тексте могут быть русские или английские слова (за слово считаем последовательность букв; внутри слова могут быть дефисы, но дефис не может идти несколько раз подряд и не может стоять на границе слова), причем кириллические буквы не могут смешиваться с латинскими, а пунктуация присутствует только стандартная (никаких испанских вопросительных знаков). 

Бонусный балл: подумайте, как можно возвращать не просто список символов, а список кортежей "токен-тип", где тип - это "русское/иностранное/пунктуация". 

In [ ]:
# your code here

#### Задача 3.

Сравните свой токенизатор с токенизатором razdel. Какой вам больше нравится? (Возьмите для проверки какой-нибудь сложный текст). 

In [ ]:
# your code here

#### Задача 4.

Возьмите сентенайзеры из DeepPavlov и razdel. Подберите достаточно сложный, по вашему мнению, текст (подсказка: можно взять какой-нибудь пост из социальной сети), протестируйте токенизацию. Какие есть различия, какой сентенайзер вам больше нравится? 

Бонус: возьмите достаточно большой текстовый файл (войну и мир, например), токенизируйте сперва одним токенизатором, а потом другим и сравните время. Как измерить затраченное время в юпитере? Можно использовать магическую функцию %%timeit. Она возвращает среднее время, затраченное на выполнение кода в ячейке. Пример использования:

In [1]:
%%timeit
lst = [i ** 0.5 for i in range(10000)]

825 µs ± 2.95 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)


Числа здесь - количество микросекунд (одна миллионная секунды). Питон сообщает, что прогнал это выражение 7 раз, каждый раз по 1000 раз (итого - 7000 раз), на каждом выполнении замерил, сколько времени это у него заняло, посчитал среднее арифметическое (825 микросекунд) и стандартное отклонение (то есть, в среднем время выполнения команды может отличаться от среднего арифметического не больше чем на 3 микросекунды). 

In [ ]:
# your code here

#### Задача 3.5

Сравните время работы для своего токенизатора и токенизатора razdel. 

In [ ]:
# your code here